crates/router/src/routes/payment_methods.rs

```diff
@@ -115,38 +115,44 @@ pub async fn list_payment_method_api(
     ),
     tag = "Payment Methods",
     operation_id = "List all Payment Methods for a Customer",
-    security(("api_key" = []), ("ephemeral_key" = []))
+    security(("api_key" = []), ("publishable_key" = []))
 )]
 #[instrument(skip_all, fields(flow = ?Flow::CustomerPaymentMethodsList))]
+#[instrument(skip_all, fields(flow = ?Flow::CustomerPaymentMethodsList))]
 pub async fn list_customer_payment_method_api(
     state: web::Data<AppState>,
-    customer_id: web::Path<(String,)>,
+    customer_id: Option<web::Path<(String,)>>,
```

Comment:
Just a ques , what will happen if we write `web::Path<(Option<String>,)>`

Reply:
then the sdk or the client would require to customer_id. where as the client will pass publishable key along with client-secret and we will fetch customer_id based on the client_secret